SIGSEGV: segmentation violation code in prometheus/parser.go:41

### Relevant telegraf.conf

```toml
[[inputs.prometheus]]
    urls = [ "http://localhost:9092/metrics/"]
    metric_version = 2
```


### Logs from Telegraf

```text
2024-03-19T08:13:37Z I! Loading config: /etc/telegraf/telegraf.conf
2024-03-19T08:13:37Z W! DeprecationWarning: Option "fielddrop" of plugin "inputs.cpu" deprecated since version 1.29.0 and will be removed in 2.0.0: use 'fieldexclude' instead
2024-03-19T08:13:37Z I! Loading config: /etc/telegraf/telegraf.d/nifi.conf
2024-03-19T08:13:37Z I! Loading config: /etc/telegraf/telegraf.d/procstat.conf
2024-03-19T08:13:37Z W! DeprecationWarning: Option "fieldpass" of plugin "inputs.procstat" deprecated since version 1.29.0 and will be removed in 2.0.0: use 'fieldinclude' instead
2024-03-19T08:13:37Z I! Starting Telegraf 1.30.0 brought to you by InfluxData the makers of InfluxDB
2024-03-19T08:13:37Z I! Available plugins: 233 inputs, 9 aggregators, 31 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-03-19T08:13:37Z I! Loaded inputs: cpu disk diskio kernel mem net processes procstat prometheus (2x) system
2024-03-19T08:13:37Z I! Loaded aggregators:
2024-03-19T08:13:37Z I! Loaded processors:
2024-03-19T08:13:37Z I! Loaded secretstores:
2024-03-19T08:13:37Z I! Loaded outputs: prometheus_client
2024-03-19T08:13:37Z I! Tags enabled: env=prod project=nifi
2024-03-19T08:13:37Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"xxxxxx", Flush Interval:10s
2024-03-19T08:13:37Z D! [agent] Initializing plugins
2024-03-19T08:13:37Z W! DeprecationWarning: Value "false" for option "ignore_protocol_stats" of plugin "inputs.net" deprecated since version 1.27.3 and will be removed in 1.36.0: use the 'inputs.nstat' plugin instead
2024-03-19T08:13:37Z I! [inputs.prometheus] Using the label selector:  and field selector:
2024-03-19T08:13:37Z I! [inputs.prometheus] Using the label selector:  and field selector:
2024-03-19T08:13:37Z D! [agent] Connecting outputs
2024-03-19T08:13:37Z D! [agent] Attempting connection to [outputs.prometheus_client]
2024-03-19T08:13:37Z I! [outputs.prometheus_client] Listening on http://[::]:9273/metrics
2024-03-19T08:13:37Z D! [agent] Successfully connected to outputs.prometheus_client
2024-03-19T08:13:37Z D! [agent] Starting service inputs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x3539de9]

goroutine 85 [running]:
github.com/influxdata/telegraf/plugins/parsers/prometheus.(*Parser).Parse(0xc002885d68, {0xc002a20000, 0x3767, 0x4000})
        /go/src/github.com/influxdata/telegraf/plugins/parsers/prometheus/parser.go:41 +0x109
github.com/influxdata/telegraf/plugins/inputs/prometheus.(*Prometheus).gatherURL(0xc0024be008, {0xc001d05950, 0xc001d05950, {0x0, 0x0}, 0x0, {0x0, 0x0}}, {0x8ac2740, 0xc0012af6c0})
        /go/src/github.com/influxdata/telegraf/plugins/inputs/prometheus/prometheus.go:496 +0x1179
github.com/influxdata/telegraf/plugins/inputs/prometheus.(*Prometheus).Gather.func1({0xc001d05950, 0xc001d05950, {0x0, 0x0}, 0x0, {0x0, 0x0}})
        /go/src/github.com/influxdata/telegraf/plugins/inputs/prometheus/prometheus.go:354 +0xfb
created by github.com/influxdata/telegraf/plugins/inputs/prometheus.(*Prometheus).Gather in goroutine 84
        /go/src/github.com/influxdata/telegraf/plugins/inputs/prometheus/prometheus.go:352 +0xd6
```


### System info

Telegraf 1.30.0 Docker 25.0.4 Nifi 1.21

### Docker

_No response_

### Steps to reproduce

1.Create a nifi prometheus endpoint (PrometheusReportingTask (org.apache.nifi - nifi-prometheus-nar))
2. Create attached telegraf configuration file
3. Run telegraf 1.30.0 with debug
4. Enjoy..
...

Cause:
nifi exporter does not output Content-Type header at all
```
>  curl -v http://localhost:9092/metrics/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9092 (#0)
> GET /metrics/ HTTP/1.1
> Host: localhost:9092
> User-Agent: curl/7.61.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Tue, 19 Mar 2024 08:30:02 GMT
< Transfer-Encoding: chunked
< Server: Jetty(9.4.51.v20230217)
```

### Expected behavior

Metrics.
Previous versions work. Tested at 1.29.5

### Actual behavior

SIGSEGV: segmentation violation code

### Additional info

_No response_

--- a/plugins/inputs/prometheus/prometheus.go
+++ b/plugins/inputs/prometheus/prometheus.go
@@ -492,6 +492,7 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) (map[s
 		Header:          resp.Header,
 		MetricVersion:   p.MetricVersion,
 		IgnoreTimestamp: p.IgnoreTimestamp,
+		Log:             p.Log,
 	}
 	metrics, err := metricParser.Parse(body)
 	if err != nil {

--- a/plugins/parsers/prometheus/parser.go
+++ b/plugins/parsers/prometheus/parser.go
@@ -38,7 +38,7 @@ func (p *Parser) Parse(data []byte) ([]telegraf.Metric, error) {
 			data = append(data, []byte("\n")...)
 		}
 	case expfmt.FmtUnknown:
-		p.Log.Warnf("Unknown format %q... Trying to continue...", p.Header.Get("Content-Type"))
+		p.Log.Debugf("Unknown format %q... Trying to continue...", p.Header.Get("Content-Type"))
 	}
 	buf := bytes.NewBuffer(data)
 	decoder := expfmt.NewDecoder(buf, format)

